src/DeviceRenderer.js

```diff
@@ -836,4 +851,24 @@ module.exports = class DeviceRenderer {
             this.emit(data.type, data.message);
         }
     }
+
+    /**
+     * Destructor for the device renderer. This won't actually destroy the instance, but simply remove all event bindings
+     * so that things can be garbage-collected.
+     * References to the instance in the caller need to be manually deleted too in order for the instance to be garbage-collected.
+     * This method also calls recursively the destroy methods on the plugins if they exist.
+     */
+    destroy() {
+        document.removeEventListener('click', this.clickHandlerCloseOverlay.bind(this));
+        this.peerConnection?.removeEventListener('connectionstatechange', this.onConnectionStateChange);
+        this.mediaManager?.destroy();
+        this.gamepadManager?.destroy();
+        this.peerConnectionStats?.destroy();
+        delete this.peerConnectionStats;
+        delete this.video;
+        delete this.wrapper;
+        delete this.videoWrapper;
+        delete this.root;
+        delete this.stream;
```

Comment:
We need to ensure that all addEventListener instances are removed before destruction, or this may lead to a memory leak.

Reply:
I'm not sure what you mean here. Isn't this what I'm doing in this function?